planner/core/logical_plan_builder.go

```diff
@@ -180,6 +180,33 @@ func (b *PlanBuilder) buildResultSetNode(node ast.ResultSetNode) (p LogicalPlan,
 	}
 }
 
+// pushDownConstExpr checks if the condition is from filter condition, if true, push it down to both
+// children of join, whatever the join type is; if false, push it down to inner child of outer join,
+// and both children of non-outer-join.
+func (p *LogicalJoin) pushDownConstExpr(expr expression.Expression, leftCond []expression.Expression,
+	rightCond []expression.Expression, filterCond bool) ([]expression.Expression, []expression.Expression) {
+	switch p.JoinType {
+	case LeftOuterJoin, LeftOuterSemiJoin, AntiLeftOuterSemiJoin:
+		if filterCond {
+			leftCond = append(leftCond, expr)
+			p.RightConditions = append(p.RightConditions, expr)
```

Comment:
Could you explain why we add it to the `p.RightConditions`, not `rightConds`? It is a little confusion here.

Reply:
confusion +1 :joy:

Reply:
Because if we append it to `rightConds`, it cannot be pushed down to children of left join, since only join condition on inner table can be pushed down for outer join.

Reply:
Better to put this in the comment?

Reply:
Comments added.